src/ConnectedMachine/readme.md

```diff
@@ -40,45 +40,108 @@ In this directory, run AutoRest:
 > see https://aka.ms/autorest
 
 ``` yaml
-powershell: true
-azure: true
-branch: master
-repo: https://github.com/Azure/azure-rest-api-specs/tree/$(branch)
-metadata:
-  authors: Microsoft Corporation
-  owners: Microsoft Corporation
-  description: 'Microsoft Azure PowerShell: $(service-name) cmdlets'
-  copyright: Microsoft Corporation. All rights reserved.
-  tags: Azure ResourceManager ARM PSModule $(service-name)
-  companyName: Microsoft Corporation
-  requireLicenseAcceptance: true
-  licenseUri: https://aka.ms/azps-license
-  projectUri: https://github.com/Azure/azure-powershell
-prefix: Az
-subject-prefix: 'Connected'
-module-name: $(prefix).$(service-name)
-namespace: Microsoft.Azure.PowerShell.Cmdlets.$(service-name)
-clear-output-folder: true
-output-folder: .
-input-file:
-  - $(repo)/specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2019-08-02/HybridCompute.json
-
+require:
+  - $(this-folder)/../readme.azure.noprofile.md
 module-version: 0.1.0
 title: ConnectedMachine
+subject-prefix: 'Connected'
+input-file:
+  - $(repo)/specification/hybridcompute/resource-manager/Microsoft.HybridCompute/stable/2019-12-12/HybridCompute.json
 
 directive:
   - where:
       subject: Operation
     hide: true
   - where: $.definitions.Identifier.properties
     suppress: R3019
-  - remove-operation: 
+
+  # GetViaIdentity isn't useful until Azure PowerShell supports piping of different subjects
+  - where:
+      verb: New
+      variant: ^GetViaIdentity\d?$
+    remove: true
+
+  # Set correct variants for PUT and PATCH verbs
+  - where:
+      verb: New
+      variant: ^CreateViaIdentity\d?$|^Create\d?$
+    remove: true
+  - where:
+      verb: Set
+      variant: ^Update\d?$|^UpdateViaIdentity\d?$
+    remove: true
+  - where:
+      verb: Update
+      variant: ^Update\d?$|^UpdateViaIdentity\d?$
+    remove: true
+
+  # New, Update, and Set all do the same thing so we hide New and Update.
+  - where:
+      verb: Update|New
+      subject: MachineExtension
+    hide: true
+
+  # Make parameters friendlier for extensions
+  - where:
+      subject: MachineExtension
+      parameter-name: Name
+    set:
+      parameter-name: MachineName
+  - where:
+      subject: MachineExtension
+      parameter-name: ExtensionName
+    set:
+      parameter-name: Name
+  - where:
+      subject: MachineExtension
+      parameter-name: PropertiesType
+    set:
+      parameter-name: ExtensionType
+  - where:
+      model-name: MachineExtension
+      property-name: PropertiesType
+    set:
+      property-name: MachineExtensionType
+  - where:
+      subject: MachineExtension
+      parameter-name: Setting
+    set:
+      alias: Settings
+  - where:
+      subject: MachineExtension
+      parameter-name: ProtectedSetting
+    set:
+      alias: ProtectedSettings
```

Comment:
Use singular name ProtectedSetting here

Reply:
It's an alias. The original parameter (which is singular) still works. Plus in this case, plural makes more sense because you are providing a hashtable which represents "the settings of the extension" that will always be plural.